src/main/java/org/opensearch/plugin/insights/core/service/categorizer/QueryShapeGenerator.java

```diff
@@ -243,11 +203,20 @@ static String buildSortShape(List<SortBuilder<?>> sortBuilderList, Boolean showF
      * @return String: comma separated list with leading space in square brackets
      * Ex: " [my_field, width:5]"
      */
-    static String buildFieldDataString(List<Function<Object, String>> methods, NamedWriteable builder) {
+    static String buildFieldDataString(ConcurrentHashMap<Class<?>, List<Method>> fieldMethodsMap, NamedWriteable builder) {
         List<String> fieldDataList = new ArrayList<>();
-        if (methods != null) {
-            for (Function<Object, String> lambda : methods) {
-                fieldDataList.add(lambda.apply(builder));
+        for (ConcurrentHashMap.Entry<Class<?>, List<Method>> entry : fieldMethodsMap.entrySet()) {
+            if (entry.getKey().isInstance(builder)) {
+                List<Method> methodsList = entry.getValue();
+                for (Method method : methodsList) {
+                    try {
+                        fieldDataList.add((String) method.invoke(builder));
+                    } catch (Exception e) {
```

Comment:
nit: can catch the specific exceptions here rather than catchall. Eg:
`catch (IllegalAccessException | InvocationTargetException e) {`

Reply:
Fixed